src/main/java/core/basesyntax/model/FruitTransaction.java

```diff
@@ -0,0 +1,64 @@
+package core.basesyntax.model;
+
+public class FruitTransaction {
+    private static final int ZERO = 0;
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        if (fruit == null || fruit.isEmpty()) {
+            throw new RuntimeException("Fruit cannot be null or empty");
+        }
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        if (quantity < ZERO) {
+            throw new RuntimeException("Quantity cannot be negative");
+        }
+        this.quantity = quantity;
+    }
```

Comment:
don't complicate the setter method. You should validate it in the PurcheseOperationHandler